ignite/src/main/java/com/yahoo/ycsb/db/IgniteYCSBClient.java

```diff
@@ -0,0 +1,181 @@
+import com.yahoo.ycsb.StringByteIterator;
+import com.yahoo.ycsb.workloads.CoreWorkload;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ *
+ */
+public class IgniteYCSBClient extends com.yahoo.ycsb.DB {
+
+  public static final String INSERT_ASYNC = "insertAsync";
+  public static final String IGNITE_IPS = "igniteIPs";
+  public static final String BATCH_SIZE = "batchSize";
+
+  private static final Logger LOG = LoggerFactory.getLogger(IgniteYCSBClient.class);
+  private static Ignite ignite;
+  private static IgniteCache<String, Map<String, String>> cache;
+
+  private boolean asyncInsert = false;
+  private int batchSize = 1;
+  private int batchCount = 0;
+  private Map<String, Map<String, String>> batchBuffer = new HashMap();
+
+
+  static synchronized void createIgniteClient(String cacheName, String igniteIPs) {
+    if (ignite == null) {
+      System.out.println("IgniteYCSBClient.createIgniteClient");
+      ignite = IgniteClient.startIgnite(igniteIPs);
+      cache = ignite.getOrCreateCache(cacheName);
+    }
+  }
+
+
+  @Override
+  public void init() throws DBException {
+    System.out.println("Init this.toString() = " + this.toString());
+    super.init();
+    Properties props = getProperties();
+    String tableName = props.getProperty(CoreWorkload.TABLENAME_PROPERTY, CoreWorkload.TABLENAME_PROPERTY_DEFAULT);
+    String igniteIPs = props.getProperty(IGNITE_IPS, "127.0.0.1,127.0.0.1:47500..47509");
+    asyncInsert = Boolean.valueOf(props.getProperty(INSERT_ASYNC, "false"));
+    batchSize = Integer.parseInt(props.getProperty(BATCH_SIZE, "1"));
+    createIgniteClient(tableName, igniteIPs);
+
+    System.out.println(
+        "" + this +  ":" +
+        "\n\tBATCH_SIZE = " + batchSize +
+        "\n\tINSERT_ASYNC = " +asyncInsert);
+
+  }
+
+  @Override
+  public Status read(String table, String key, Set<String> fields,
+                     Map<String, ByteIterator> result) {
+    LOG.debug("IgniteYCSBClient.read:{}", key);
+    Map<String, String> r = cache.get(key);
+    if (r == null) {
+      return Status.NOT_FOUND;
+    } else {
+      populateResult(result, fields, r);
+      return Status.OK;
+    }
+  }
+
+  private void populateResult(Map<String, ByteIterator> sink, Set<String> fields, Map<String, String> source) {
+    if (fields == null) {
+      for (Map.Entry<String, String> entry : source.entrySet()) {
+        sink.put(entry.getKey(), new StringByteIterator(entry.getValue()));
+      }
+    } else {
+      for (String fName : fields) {
+        sink.put(fName, new StringByteIterator(source.get(fName)));
+      }
+    }
+  }
+
+  @Override
+  public Status scan(String table, String startkey, int recordcount,
+                     Set<String> fields,
+                     Vector<HashMap<String, ByteIterator>> result) {
+    LOG.warn("Ignite does not support scan semantics");
+    return Status.OK;
```

Comment:
should be `Status.NOT_IMPLEMENTED` then...

Reply:
Thanks, I used Status.OK because that is return status for similarly not supported method in Infinispan binding. Will fix.